engine/class_modules/sc_death_knight.cpp

```diff
@@ -5993,34 +5852,30 @@ struct festering_strike_t : public death_knight_melee_attack_t
       if ( rng().roll( p() -> talent.unholy.feasting_strikes -> effectN( 1 ).percent() ) )
       {
         p() -> replenish_rune( as<unsigned int>( p() -> spell.feasting_strikes_gain -> effectN( 1 ).base_value() ), p() -> gains.feasting_strikes );
-
-        if( p() -> is_ptr() )
-        {
-          p() -> trigger_runic_corruption( p() -> procs.fs_runic_corruption, 0, 1.0, true );
-        }
+        p() -> trigger_runic_corruption( p() -> procs.fs_runic_corruption, 0, 1.0, true );
       }
     }
   }
 
-    double composite_da_multiplier( const action_state_t* state ) const override
-    {
-      double m = death_knight_melee_attack_t::composite_da_multiplier( state );
-
-      if ( p() -> talent.unholy.reaping.ok() && target -> health_percentage() < p() -> talent.unholy.reaping -> effectN( 2 ).base_value() )
-      {
-        m *= 1.0 + p() -> talent.unholy.reaping -> effectN( 1 ).percent();
-      }
+  double composite_da_multiplier( const action_state_t* state ) const override
+  {
+    double m = death_knight_melee_attack_t::composite_da_multiplier( state );
 
-      return m;
+    if ( p() -> talent.unholy.reaping.ok() && target -> health_percentage() < p() -> talent.unholy.reaping -> effectN( 2 ).base_value() )
+    {
+      m *= 1.0 + p() -> talent.unholy.reaping -> effectN( 1 ).percent();
     }
-  };
+     
+    m *= 1.0 + p() -> talent.unholy.improved_festering_strike -> effectN( 1 ).percent();
```

Comment:
Since this multiplier is always active if the talent is active should this be in the constructor?